### PR TITLE
Remove CUDA support relying on abandoned toolchain

### PR DESCRIPTION
**Pull request by Ralith**
Fixes https://github.com/dimforge/nalgebra/issues/1330. Required to get CI green again. This is a breaking change.

### CODE REVIEW

**Review by tpdickso — APPROVED** [review]
Probably should wait for a more tenured maintainer to approve before merging this, but leaving my approval here to mark that I looked over the code here and it looks correct to me.

**Comment by Ralith** [comment]
Meh, CI is broken and sebcrozet's already approved the idea. Let's get green.

**Pull request merged**
